### DISCUSSION TITLE
CRAN task view: epidemiology

### DISCUSSION THREAD

**Discussion by sbfnk** [WHO-Collaboratory/collaboratory-epipipeline-community#6] [Ideas]
Identify tools, resources and potential blockers to facilitate activities on day 2 & 3.

**Comment by sbfnk** [comment]
The objective of this task is to adapt the CRAN task view for epidemiology to provide a more accessible resource for finding out which R packages are available for which epidemiological task.

Tools:
- (R)markdown, github pages, ...

Resources:
- The existing task view provides a good potential starting point

Potential blockers:
- no license in the CRAN repo
- limitations in what can be visualised as long as this lives in the CRAN task view
- need to avoid overlap with the Epi R Handbook
- not coding so potentially not good use of our time on the two days

Expected outcomes:
- to be decided: either an amended version of the existing task view or something that can meaningfully live besides it

**Reply by JamesABaker** [reply]
Link to cran repo: https://github.com/cran-task-views/Epidemiology

and the cran repo: https://cran.r-project.org/web/views/Epidemiology.html

**Comment by Bisaloo** [comment]
We are working towards an automated health check of the universe of Epidemiology R packages as defined in the CRAN Task View.

Development is ongoing at: https://github.com/Bisaloo/cran-task-view-analysis

**Reply by Bisaloo** [reply]
First working draft is live at https://hugogruson.fr/cran-task-view-analysis/ (to be moved to an organization GitHub repository later)

**Comment by JamesABaker** [comment]
We are also doing a check of as many repos as possible and adding GitHub Urls to the cran description. In the long term, this will help automate the "health check" once individual projects merge the PR.


## PR example

> The github URL is added to the description. This will give cran users awareness of the GitHub location.
>
> From a technical point of view, this is useful for allowing autolinking in pkgdown websites. Here is the explanation: https://pkgdown.r-lib.org/articles/linking.html?q=cross#across-packages
>
> For an example see https://github.com/tidyverse/dplyr/blob/main/DESCRIPTION

**Comment by JamesABaker** [comment]
# 📅 Day 1 recap

Contributors @Bisaloo @chathuracse

- Roadmap for repo health check
- Not all Cran entries were trivially associated with a GH repo. We tried to bolster this list for test purposes.
    - 🤝  Opened 10 PRs across the Cran ecosystem to include Git repos in Cran descriptions. 3 merged, 3 `wontfix` due to read only.
        - https://github.com/tobadia/R0/pull/17
        - https://github.com/reconhub/epicontacts/pull/134/
        - https://github.com/cran/epiDisplay/pull/1
        - https://github.com/reconhub/epiflows/pull/47
        - https://github.com/EU-ECDC/EpiReport/pull/17
        - https://github.com/reconhub/incidence/pull/133
        - https://github.com/cran/powerSurvEpi/pull/1
        - https://github.com/reconverse/i2extras/pull/18
        - https://github.com/reconhub/epitrix/pull/32
        - https://github.com/cran/epitools/pull/1
- 🚀 New features for repo health analytics.
    - GH API calls https://github.com/Bisaloo/cran-task-view-analysis/pull/3
    - Cran scrape (PR WIP)
    - Linting issue ⚠️ working with larger package https://github.com/bernadette-eu/Bernadette/pull/10
    - md5 comparison for workflow files https://github.com/Bisaloo/cran-task-view-analysis/pull/4
    - File presence checks https://github.com/Bisaloo/cran-task-view-analysis/pull/3
        - `pkgdown` yamls
        - readme
        - workflow folder
        - license

**Comment by JamesABaker** [comment]
# Day 2 morning

## 🚀 Features developed.

- Analytics of pass-fails https://github.com/Bisaloo/cran-task-view-analysis/pull/11 📊
- Resolved and merged API https://github.com/Bisaloo/cran-task-view-analysis/pull/3 and https://github.com/Bisaloo/cran-task-view-analysis/pull/5 🤝
- Added fingerprint function https://github.com/Bisaloo/cran-task-view-analysis/pull/4 
- MIT license added https://github.com/Bisaloo/cran-task-view-analysis/issues/1 📃

## 🛑 Issues encountered.

- Pre-commit hooks are not working https://github.com/Bisaloo/cran-task-view-analysis/issues/12
- Vectors should be implemented rather than functions, loops and OOP https://github.com/Bisaloo/cran-task-view-analysis/issues/8
- GH action can be checked directly https://github.com/Bisaloo/cran-task-view-analysis/issues/7
- Many more features on roadmap https://github.com/Bisaloo/cran-task-view-analysis/issues/2

**Reply by Bisaloo** [reply]
Results and summary plots are displayed via a quarto dashboard available at http://hugogruson.fr/cran-task-view-analysis/.

The dashboard is automatically re-rendered and deployed after each push on `main` via a GitHub Actions workflow.

**Reply by JamesABaker** [reply]
This directly shows how this resource will answer our two questions:
 - Which packages are the most complete?
 - What is the overall health & completeness of the epi package ecosystem?

**Comment by PatriciaRose963** [comment]
# Comments from reviewers

## General comments

1. The team has done a lot of great work flagging issues and making pull requests for existing packages to improve the overall health of the epi pipeline ecosystem.

2. The finalised report is likely most useful for the package developer community. It is not clear how unaware developers would be of the fact that their packages fail the presented checks.

3. For package users, this likely would not influence decision making on which packages to use - this is primarily determined by functionality.

4. We were confused about the deprecated deps check - would this cause the package to become archived in CRAN?

## Future directions

1. Integration of checks into the existing task view or into a platform that lists functionality and categorises packages by into subtopics would be helpful

2. It would be helpful to understand what proportion of epi packages are not captured through the epidemiology CRAN tag (github or improperly tagged CRAN packages)

**Reply by JamesABaker** [reply]
Great suggestions! Thanks very much. 

Hopefully our changes in https://github.com/Bisaloo/cran-task-view-analysis/pull/16 and https://github.com/Bisaloo/cran-task-view-analysis/pull/15 go some way to explaining our vision for the long term support of users as well as maintainers. They also should help define what each check is useful for as a proxy for quality.

The future directions suggestions are closely aligned to some of the issues on our roadmap, so it is good to hear that we are on the right track!
